### PR TITLE
added missing flush/close of reader and writer (repairs failing unit-tests)

### PR DESCRIPTION
**Pull request by afischer211**
Try to build the plugin locally, but get failing unit-tests (with Windows10, Java8u181, Maven3.5).
After adding close for reader and writer, the unit-tests are working again.

This pull request makes the following changes:
* added missing flush/close of reader and writer (repairs failing unit-tests)

It relates to the following issue #s:
* failing unit-tests

### CODE REVIEW

File: src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java [hunk 1]
```diff
@@ -94,6 +94,10 @@ public TempBlob removeMirrorUrlFromArtifactsXml(final TempBlob artifact,
             XMLEventWriter writer = outputFactory.createXMLEventWriter(xmlOut);
 
             streamXmlToWriterAndRemoveAbsoluteUrls(reader, writer);
+			
+            reader.close();
```
[the remark this answers is not in the document]
Comment (jlstephens89) [reply src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java]: @afischer211 Really appreciate you fixing this. We prefer to have PRs be focused to a single change, could you split out the two modifications into two separate PRs please.

**Comment by laeubi** [comment]
This seems to be a similar problem as described in https://github.com/sonatype-nexus-community/nexus-repository-p2/pull/10

**Comment by DarthHater** [comment]
@laeubi please split out the Docker modifications into another PR, @jlstephens89 is right (write?) on that :)

**Comment by DarthHater** [comment]
I went ahead and merged #10 , if you want to contribute the Docker stuff I'd throw that into another PR! Thanks as always, @laeubi !

File: CONTRIBUTORS.md [hunk 2]
```diff
@@ -24,4 +24,6 @@ Sonatype internal people:
 
 External contributors:
 
+* [@mgh87](https://github.com/mgh87/) (Martin Huter)
```
[the remark this answers is not in the document]
Comment (jlstephens89) [reply CONTRIBUTORS.md]: Different user?

**Comment by laeubi** [comment]
@DarthHater I just commented on this I don't created the PR ;-) But thanks for mergeing this, is there already a version published that contains the fix?

**Comment by DarthHater** [comment]
Yep! Should be at: https://github.com/sonatype-nexus-community/nexus-repository-p2/releases

**Pull request closed without merging**